File: S3Source/typescript/lib/app-stage.ts
/**
 * Deployment stage for pipeline.
 */
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { AppStack } from './app-stack';

export class AppStage extends cdk.Stage {
  constructor(scope: Construct, id: string, props?: cdk.StageProps) {
    super(scope, id, props);

    new AppStack(this, 'AppStack');
  }
}

File: S3Source/csharp/src/AppStage.cs
using Amazon.CDK;
using Constructs;

namespace S3Source
{
    public class AppStage : Stage
    {
        internal AppStage(Construct scope, string id, IStageProps props = null) : base(scope, id, props)
        {
            new AppStack(this, "AppStack");
        }
    }
}
